### DISCUSSION TITLE
Problem building in Windows 10, Visual Studio 2019

### DISCUSSION THREAD

**Discussion by fred26** [rizinorg/rizin#1932] [Archive] [answered]
I am having problems building via meson in Windows. I run `meson --buildtype=release --prefix=%CD%\rizin-install build`  from the "x64 Native Tools Command Prompt for VS 2019" and receive the following errors:

> The Meson build system
Version: 0.59.4
Source dir: D:\Users\feder\Files\Code\Development\rizin-dev
Build dir: D:\Users\feder\Files\Code\Development\rizin-dev\build
Build type: native build
Project name: rizin
Project version: v0.4.0-git
C compiler for the host machine: cl (msvc 19.27.29111 "Microsoft (R) C/C++ Optimizing Compiler Version 19.27.29111 for x64")
C linker for the host machine: link link 14.27.29111.0

>meson.build:1:0: ERROR: Unknown linker(s): [['lib'], ['llvm-lib']]
The following exception(s) were encountered:
Running "lib /?" gave "[WinError 193] %1 is not a valid Win32 application"
Running "llvm-lib /?" gave "[WinError 2] The system cannot find the file specified"

The command `lib /?` is obviously working from the "x64 Native Tools Command Prompt for VS 2019"

Where can I edit the meson script to fix this? Any other comment on how to make this work?

Thanks!

**Comment by XVilka** [comment]
It should work. We use `dist/windows/vsdevenv.ps1` and `dist/windows/build_windows_installer.ps1` scripts. Maybe you could try those instead from the general PowerShell?

**Comment by ret2libc** [comment, marked as answer]
```
Running "lib /?" gave "[WinError 193] %1 is not a valid Win32 application"
Running "llvm-lib /?" gave "[WinError 2] The system cannot find the file specified"
```
I suggest to update meson first and make sure your windows installation is in a good shape. Never seen this error and it does not seem strictly related to Rizin.
